pkg/accesscontrol/policy_rule_index.go

```diff
@@ -93,19 +93,29 @@ func addResourceAccess(accessSet *AccessSet, namespace string, rule rbacv1.Polic
 	for _, group := range rule.APIGroups {
 		for _, resource := range rule.Resources {
 			names := rule.ResourceNames
-			if len(names) == 0 {
-				names = []string{All}
+			if namespace != All && resource == "namespaces" && group == "" {
```

Comment:
Maybe add a comment here to help our future selves understand what is going on here.

Reply:
Yes, what's so special about `namespace != All` that you need to set `names = []string{namespace}` ?  What if `names` is set to a non-empty array of some other value? Can it just bet thrown out?

Reply:
I added code comments and changed the code. please re-review.